### ISSUE TITLE
[BUG] Tab supposed to be in light mode reverts to dark mode itself when switching between tabs

### ISSUE THREAD

**Issue by tddschn** [darkreader/darkreader#4215]
### **Describe the bug**

I had 2 tabs `tabA` and `tabB` open, and they were on different domains. In `tabA`, I disabled dark mode for the current domain by pressing opt-shift-A, so it appeared in the original color. Then I swtiched to `tabB`, which was still in dark mode, then I switched back to `tabA`, it appeared in dark mode too! However, the domain for `tabA` is still among the list that dark mode did not apply. To disable dark mode for `tabA`, I had to press opt-shift-A twice or clicking `Toggle current website` button twice.

### **Steps to reproduce**

The bug appears randomly bug quite frequently. 

The last time I noticed it, I had a dozens of tabs open in Google Chrome 87 on macOS Big Sur (AFAIK, this bug affects lower version of macOS and Chrome on GNOME too). 

[`TabA`](https://www.pmalerts.com/) was `https://www.pmalerts.com/` and [`tabB`](https://news.ycombinator.com/item?id=25161117) was `https://news.ycombinator.com/item?id=25161117`. The two tabs are the 4th and 5th tabs counting from the right of the tab bar. I disabled dark mode on [`tabA`](https://www.pmalerts.com/), and did not change settings for [`tabB`](https://news.ycombinator.com/item?id=25161117). I was in `tabA`, and then `control-tab`ed once to go to `tabB` and then `control-shift-tab`ed back to the tab, and it appeared in dark mode even if the extension showed that dark mode was disabled. And I had to press opt-shift-A twice or clicking `Toggle current website` button twice to actually disable dark mode for this site.

### **Expected behavior**
`TabA` is supposed to always be in light mode after I set it to do so.

### **Actual behavior**

See `Steps to reproduce` above.
Like I said, the bug appears randomly bug quite frequently. 

### **Screenshots**
TabA (set to light mode but appears in dark mode)

![image](https://user-images.githubusercontent.com/45612704/99863972-b1750300-2bdb-11eb-8578-4a0bee682063.png)

TabB:

![image](https://user-images.githubusercontent.com/45612704/99863993-d2d5ef00-2bdb-11eb-82e6-719cc42f84a8.png)

### **System Information:**

- OS: macOS Big Sur 10.16 20B29 x86_64
- Browser: Google Chrome 87.0.4280.67
- Dark Reader version: 4.9.24

**Comment by Gusted** [comment]
Hi @tddschn!

Do you have anything else in your sitelist related to the site?
Also does this happen when don't use the shortcut but instead press on the site toggle.

Regards,
Gusted

**Comment by tddschn** [comment]
I don't think I have anything else in the `Not invert listed` list that relates to pmalerts.com. I was visiting the site for the first time.

I tried to reproduce the bug but I failed... Next time I notice the bug I'll try to use the extension button to toggle, and maybe attach a screencast. :)

**Comment by tddschn** [comment]
I made a gif from a screen recording but it loops on GitHub, making it difficult to understand.

The recording is also available in `mp4` in the zip file. 👍 

![presenta-darkreader-bug](https://user-images.githubusercontent.com/45612704/101234194-49084480-36f8-11eb-895d-cd7c86e77da6.gif)

[presenta-darkreader-bug.mp4.zip](https://github.com/darkreader/darkreader/files/5646501/presenta-darkreader-bug.mp4.zip)

**Comment by Gusted** [comment]
Hi @tddschn!

It's weird how this happens, as Dark Reader only execute `removing` code on the site if it's disabled to `clean-up` some things. It got me thinking that Chrome somehow got Dark Reader `executed` again, could you next time this happens do:
Type in the console `document.querySelectorAll('.darkreader');` and see what is being returned back?

Regards,
Gusted

**Comment by tddschn** [comment]
@Gusted Thank you! I'll do that next time. :)

**Comment by Gusted** [comment]
^ I ask you to do that as I might suspect that the `Fallback style` is being applied again somehow and I suspect because it's not a `new tab` the javascript to remove that style will never be executed.

**Comment by joylazari** [comment]
Hey, I have the same issue.
I first noticed with Slack for web.
I'm using Chrome 87.0.4280.88 on macOS 10.15.6

The site is disabled and present in the "ignore" list and the NodeList is an empty array.
But when I move away (another app, change tab ...) and then come back to the tab, the dark mode is then applied again with the following result in the console:

![Screenshot 2020-12-11 at 14 30 26](https://user-images.githubusercontent.com/6898095/101909356-a8b19480-3bbd-11eb-8b32-f262f9bae314.png)

**Comment by Gusted** [comment]
Sometimes I have just a little feeling that browsers are the problem. And I might give it a wild guess the same is happening here as switching tabs doesn't trigger Dark Reader in any way after the theme is applied.

**Comment by Gusted** [comment]
> @Gusted here's a screenshot of the same query. It seems like this confirms your theory?

Nope, this lets me more think that Google Chrome has and edge case with some of our internals. Time to boot up old boy Google Chrome and see if I can get an consistent reproduce-able case.

**Comment by tmandry** [comment]
Okay, let me know if any more queries will help. This doesn't seem to happen all the time.. I think the last change I made was enable auto mode. It went dark in the evening, I woke up the laptop the next day and some sites were still dark. But it also might have nothing to do with that 🤷

**Comment by Gusted** [comment]
> But it also might have nothing to do with that shrug

Might been chrome suspending the tabs while keeping the tab still in some storage and when accesing it again it will return to that state..... however their seems to be that the connected extensions states are not being saved. IDK what chrome has been doing lately, but it seems to be an regression while I remembered some similar story's a while back mid-June.

**Comment by tmandry** [comment]
For anyone who's experiencing this, toggling the extension off and on again in extension manager fixes the problem for me, temporarily at least.

**Comment by xss** [comment]
I just experienced what appears to be the same issue, but on Firefox: Upon tab switching, Dark Reader suddenly started to apply dark mode to sites that are on the "Not invert listed" list. Removing and re-adding the site or disabling and re-enabling Dark Reader restores the site's intended display – until the next tab switch.

It seems this is not only an issue in the Chrome version of the extension.

> For anyone who's experiencing this, toggling the extension off and on again in extension manager fixes the problem for me, temporarily at least.

Thank you, @tmandry, for this hint. This works in Firefox, as well!
